### PR TITLE
Cmake-ify and Modernize

### PR DESCRIPTION
**Pull request by spiderkeys**
There are still a number of unfixed warnings of the form:

```
warning: implicitly-declared ‘constexpr XsFilterProfile& XsFilterProfile::operator=(const XsFilterProfile&)’ is deprecated [-Wdeprecated-copy]
```

I tried to fix these by implementing what I thought was the correct constructor implementation, but I got a SIGSEGV for certain types:
```c++
XsFilterProfile& operator=(const XsFilterProfile& other) {
	return *this = XsFilterProfile(other);
}
```

### CODE REVIEW

**Comment by gilborty** [comment]
Just poking around, and I think you need to copy each member? Like check for self assignment and then copy each member over since it is overloading the operator? 

Another thing is that since they are defining these, I wonder if we need a move constructor / operator as well (or do they have it?)

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.15)
+project(xsens_examples C CXX)
```
Comment (gilborty) [inline examples/CMakeLists.txt:2]: Might want to put in the README that we need at least:

```
 sudo apt-get install libusb-dev
```

**Comment by spiderkeys** [comment]
All warnings should now be properly resolved. Project structure has been set up to be directly included as a remote, plain-cmake package that can be built with colcon.

**Review by gilborty — APPROVED** [review]
Builds without errors for me. Even though `cppcheck` has some ideas on how to make it less bug prone. Thanks for doing this.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (spiderkeys) [reply examples/CMakeLists.txt:2]: The dependency is specified via package.xml, so it should get pulled in with the normal rosdep install. I will make a note in the readme for non-ROS builds though.

**Pull request merged**